Modify action explanations to reflect path parameters

helfi92 opened the pull request:
The best option here is to switch actions back to allowing substitution in the URL path. This pull-request modifies the action explanations to talk about path parameters.

helfi92 commented:
I will modify the docs in a different pull-request.

djmitche commented on schemas/constants.yml:
```diff
@@ -132,25 +132,33 @@ dependency-relation:
-  perform a `POST` request. Note that the request will be signed with the
-  user's Taskcluster credentials.
+  Actions are defined on the provisioner and they are performed via a `POST` request using the `url`.
+  Depending on the `context`, the `url` will be able to fill path parameters. In addition, `context` is
```
..will be filled with parameters

djmitche commented on schemas/constants.yml:
```diff
@@ -132,25 +132,33 @@ dependency-relation:
+
+  | `context`   | Path parameters                                      | Page displayed        |
+  |-------------|------------------------------------------------------|-----------------------|
+  | provisioner | :provisionerId                                       | Worker explorer       |
```
Provisioner Explorer

djmitche replied:
Also, you can probably omit the path parameters here, as they are included in action-url-description.

djmitche commented on schemas/constants.yml:
```diff
@@ -132,25 +132,33 @@ dependency-relation:
+  | `context`   | Path parameters                                      | Page displayed        |
+  |-------------|------------------------------------------------------|-----------------------|
+  | provisioner | :provisionerId                                       | Worker explorer       |
+  | worker-type | :provisionerId, :workerType                          | Workers explorer      |
```
Worker-Type Explorer

djmitche commented on schemas/constants.yml:
```diff
@@ -132,25 +132,33 @@ dependency-relation:
+  |-------------|------------------------------------------------------|-----------------------|
+  | provisioner | :provisionerId                                       | Worker explorer       |
+  | worker-type | :provisionerId, :workerType                          | Workers explorer      |
+  | worker      | :provisionerId, :workerType, :workerGroup, :workerId | Worker types explorer |
```
Worker Explorer

djmitche commented on schemas/constants.yml:
```diff
@@ -132,25 +132,33 @@ dependency-relation:
-  and `context` property are used to make the `POST` request.
-  The request needs to be signed with the user's Taskcluster credentials.
+  When an action is triggered, a `POST` request is made using the `url`.
+  Depending on the `context`, the `url` will be able to fill the following path parameters:
```
..the following parameters will be substituted in the `url`:

jonasfj commented on test/workerinfo_test.js:
```diff
@@ -117,13 +117,13 @@ suite('provisioners and worker-types', () => {
         title: 'Kill Worker Type',
         context: 'worker-type',
-        url: 'https://hardware-provisioner.mozilla-releng.net/v1/power-cycle',
+        url: 'https://hardware-provisioner.mozilla-releng.net/v1/power-cycle/:provisionerId/:workerType',
```
Against please consider `<provisionerId>` instead of `:provisionerId` as the parameter format... this is more in line with what we do in `references.json`..

helfi92 commented:
@jonasfj @djmitche  I added a method property to the action definition. I also removed `format: uri` from the action `url` since we decided to use something like `<provisionerId>` instead of `:provisionerId`.

djmitche commented on schemas/list-workers-response.yml:
```diff
@@ -85,7 +85,12 @@ properties:
+          description: |
+            Method to indicate the desired action to be performed for a given resource.
+          type:   string
+          enum:   ["GET", "POST", "PUT", "DELETE", "PATCH"]
```
I don't think GET should ever perform an action.

helfi92 replied:
This must have slipped in by accident. Fixing that right now!

helfi92 commented:
@jonasfj I will merge now but if there's anything you see that needs to be added/removed, let me know :)
